### PR TITLE
Feat support bucket name

### PR DESCRIPTION
**Pull request by Juiced66**
## What does this PR do?
Add `bucketName` api arg support 

It's now possible to override bucketName base config

### How should this be manually tested?

  - Step 1 : add `bucketName`  arg to an api call

### Other changes
Update doc

### CODE REVIEW

**Comment by Juiced66** [comment]
Is packages-lock.json update a problem ?

File: README.md [hunk 1]
```diff
@@ -231,6 +231,15 @@ Then in your `kuzzlerc` file, you can change the following configuration variabl
   }
 }
 ```
+If needed, you can override `bucketName` by adding `bucketName` arg to the request:
```
Comment (rolljee) [inline README.md:234]: You need to update the doc/2 documentation accordingly with the new parameters

**Review by rolljee — CHANGES_REQUESTED** [review]
Also this lack of tests

**Comment by rolljee** [comment]
> Is packages-lock.json update a problem ?

This is to me not a problem 😄

File: lib/S3Plugin.js [hunk 2]
```diff
@@ -162,10 +162,11 @@ class S3Plugin {
 
     const filename = this._stringArg(request, 'filename'),
       uploadDir = this._stringArg(request, 'uploadDir'),
+      Bucket = this._stringArg(request, 'bucketName', this.config.bucketName),
```
[the remark this answers is not in the document]
Comment (Juiced66) [reply lib/S3Plugin.js:165]: Yes it does

File: README.md (hunk 1 above)
Comment (Juiced66) [reply README.md:234]: It is done

**Comment by Juiced66** [comment]
> Also this lack of tests

not anymore :D

File: lib/S3Plugin.js [hunk 3]
```diff
@@ -162,20 +162,21 @@ class S3Plugin {
...
```
Comment (Aschen) [inline lib/S3Plugin.js:165]: Variables should not start with a capital letter

**Comment by Juiced66** [comment]
I'll close this PR : 
- I had merged it with [the PR for bucketController](https://github.com/kuzzleio/kuzzle-plugin-s3/pull/26) has it was necessary to merge this one first

File: lib/S3Plugin.js (hunk 3 above)
Comment (Juiced66) [reply lib/S3Plugin.js:165]: I've done changes and I've merged this PR onto the otherer see closing comment

**Pull request closed without merging**